Merge particleSystem into main

godofocean7 opened the pull request.

godofocean7 committed:
blood splash particle system added

godofocean7 committed:
Blood splash particle system added, level three edited

godofocean7 committed:
popup message added

godofocean7 committed:
Merge pull request #48 from matthewwarn/main

Merge main to particle System

SkyePooley reviewed (changes requested):
Blood splash and rain particle effects look great and I hope to have those integrated with the attack animations soon.

Level three has not followed level creation procedure and is therefore incompatible with our backend. This must be changed before merging.

godofocean7 committed:
Modified blood rain particle system

godofocean7 committed:
popup messages updated

godofocean7 committed:
popup messages updated 2

SkyePooley committed:
Linked Level Three

Corrected the level linking for level three.
Moved level transition in level two as it was pushing the player to the
previous level as soon as it loaded.

SkyePooley committed:
Merge branch 'main' into particleSystem

Merged conflicts manually. I have playtested the conflicted levels after
merging and they appear to be working normally.
There may be some missing nodes as I was working with unnames resoures
in tscn files.

 Conflicts:
	Godot Project/Scenes/Levels/Level One/level_one.tscn
	Godot Project/Scenes/Levels/Level Two/level_two.tscn
	Godot Project/Scenes/Levels/Level Two/unlock_dash_zone.gd
	Godot Project/Scenes/Levels/Tutorial/integrated_tutorial.tscn
	Godot Project/Scenes/game_manager.tscn

SkyePooley committed:
Refactored level three to template

Yeoungjun didn't create level three according to procedure so it was not
compatible with the game backend.
He said it was refactored to inherit the template but it still wasn't
quite right so I copied all the contents into a new level inheriting the
template.

Also added check to signal auto linking that prevents linking to
previously linked signals.

SkyePooley reviewed (approved):
Particles look good but we will need to do some additional work in order to connect these to the enemy damage.

Level three is working now that I have transferred all the content into a fresh scene.

SkyePooley committed:
Reduced attack range of floowmaw

The attack range of the floor maw made it impossible to complete level
three so I have reduced this slightly.
